NIX.md

```diff
@@ -0,0 +1,138 @@
+# BDK's Nix Guidelines
+
+This document outlines:
+
+1. What is Nix and how to install.
+1. BDK's Nix-based tests and CI.
+1. Developer best-practices with Nix.
+1. How to run replicate CI tests locally.
+
+## Nix
+
+We use [Nix](https://nixos.org/) as the CI tool for BDK and also as a
+development environment for the project.
+Nix is purely functional.
+Everything is described as an expression/function,
+taking some inputs and producing deterministic outputs.
+This guarantees reproducible results and makes caching everything easy.
+Nix expressions are lazy. Anything described in Nix code will only be executed
+if some other expression needs its results.
+This is very powerful but somewhat unnatural for developers not familiar
+with functional programming.
+
+There are several resources to get started with Nix:
+
+- [NixOS Wiki](https://nixos.wiki/).
+- [Nix Reference Manual](https://nixos.org/manual/nix/stable/).
+- [`nixpkgs` Manual](https://nixos.org/manual/nixpkgs/stable/).
+- [Official documentation for getting things done with Nix](https://nix.dev/).
+
+There's also the [`nix-bitcoin` project](https://nixbitcoin.org/).
+
+To install Nix please follow the [official Nix documentation](https://nixos.org/manual/nix/stable/installation/installation.html).
+You may need to enable Flakes support, check the instructions at [NixOS Wiki entry on Flakes](https://nixos.wiki/wiki/Flakes).
+
+## Why are we using Nix
+
+We want to make BDK's development easier.
+This means two things:
+
+1. Enhance developer experience.
+1. Facilitate onboarding of new contributors.
+
+Both of these can be accomplished with Nix.
+
+BDK has many crates in the workspace,
+and proper development and testing needs dependencies to be installed,
+and environment variables to be set.
+This can be daunting for new contributors.
+
+BDK needs several Rust versions with different compilation targets.
+This can be cumbersome to install, but also needs proper maintainability,
+since these versions need to be updated frequently.
+
+BDK enforces commit styles and commit policies (please check [CONTRIBUTING.md](CONTRIBUTING.md)).
+This is difficult to enforce locally
+and can be a source of frustration during contribution reviews,
+potentially leading to a lot of wasted time and pushing new contributors away.
+
+Finally, BDK has a lot of tests and checks.
+It is difficult to replicate these locally.
+
+All the above can easily be accomplished with Nix.
+Nix is available for macOS, Linux, and Windows;
+while also being easy to install.
+Nix has a rich [community](https://nixos.org/community/) and one can find help
+on the [Nix's Forums](https://discourse.nixos.org/),
+[Nix's Discord](https://discord.gg/RbvHtGa),
+and [Nix's Matrix channel](https://matrix.to/#/#community:nixos.org).
+Additionally, Nix issues and errors can be searched in any search engine,
+and in the [Nix's stackoverflow](https://stackoverflow.com/questions/tagged/nix+or+nixpkgs+or+nixos+or+nixops).
```

Comment:
@storopoli just throwing some suggestions, maybe mentioning https://zero-to-nix.com/ as well ?

Reply:
I've added rebased and squashed. Thanks for the suggestion. Great resource.

Reply:
Great, thanks!